cli/.js/src/main/scala/aqua/FuncCompiler.scala

```diff
@@ -123,28 +123,53 @@ class FuncCompiler[F[_]: Files: AquaIO: Async](
       }
   }
 
+  private def compileToContext(
+    path: Path,
+    imports: List[Path],
+    config: AquaCompilerConf = AquaCompilerConf(transformConfig.constantsList)
+  ) = {
+    val sources = new AquaFileSources[F](path, imports)
+    CompilerAPI
+      .compileToContext[F, AquaFileError, FileModuleId, FileSpan.F](
+        sources,
+        SpanParser.parser,
+        config
+      )
+      .map(_.leftMap(_.map(_.show)))
+  }
+
+  private def compileBuiltins() = {
+    for {
+      path <- PackagePath.builtin.getPath()
+      context <- compileToContext(path, Nil)
+    } yield {
+      context
+    }
+  }
+
   // Compile and get only one function
   def compile(
     func: CliFunc,
-    jsonServices: List[JsonService]
+    jsonServices: List[JsonService],
+    withBuiltins: Boolean = false
   ): F[ValidatedNec[String, (FuncArrow, List[Service])]] = {
     for {
       prelude <- Prelude.init[F](withRunImport)
-      sources = new AquaFileSources[F](input, prelude.importPaths ++ imports)
-      // compile only context to wrap and call function later
-      compileResult <- Clock[F].timed(
-        CompilerAPI
-          .compileToContext[F, AquaFileError, FileModuleId, FileSpan.F](
-            sources,
-            SpanParser.parser,
-            AquaCompilerConf(transformConfig.constantsList)
-          )
-          .map(_.leftMap(_.map(_.show)))
-      )
+      // compile builtins and add it to context
+      builtinsV <-
+        if (withBuiltins) compileBuiltins()
+        else validNec[String, Chain[AquaContext]](Chain.empty).pure[F]
```

Comment:
I suggest to avoid having `withBuiltins` flag. Looks very artificial to me. 

Instead, I'd suggest to check if `importPaths ++ imports` contain `@fluencelabs/aqua-lib/builtin.aqua`, and then add `PackagePath.builtin` only if they're absent in imports.